--- curl-7.36.0/configure.ac.orig	2014-03-30 18:33:06.758735667 +0200
+++ curl-7.36.0/configure.ac	2014-03-30 20:11:15.498488539 +0200
@@ -2767,7 +2767,7 @@
   CLEANCPPFLAGS="$CPPFLAGS"
   CLEANLIBS="$LIBS"
 
-  h2pcdir=${want_h2_path}/lib/pkgconfig
+  h2pcdir="${want_h2_path:+${want_h2_path}/lib${libsuff}/pkgconfig}"
   CURL_CHECK_PKGCONFIG(libnghttp2, $h2pcdir)
 
   if test "$PKGCONFIG" != "no" ; then
