--- configure.ac.old	2005-10-15 22:33:18.000000000 +0200
+++ configure.ac	2005-10-15 22:51:37.000000000 +0200
@@ -729,13 +729,6 @@
   fi
   CPPFLAGS="$CPPFLAGS $GSSAPI_INCS"
 
-  AC_CHECK_HEADER(gss.h,
-    [
-      dnl found in the given dirs
-      AC_DEFINE(HAVE_GSSGNU, 1, [if you have the GNU gssapi libraries])
-      gnu_gss=yes
-    ],
-    [
       dnl not found, check Heimdal
       AC_CHECK_HEADER(gssapi.h,
         [
@@ -757,8 +750,6 @@
           )
         ]
       )        
-    ]
-  )
 else
   AC_MSG_RESULT(no)
 fi
