diff -Nru curl-7.9.1.org/lib/ftp.c curl-7.9.1/lib/ftp.c
--- curl-7.9.1.org/lib/ftp.c	Thu Nov  1 13:47:22 2001
+++ curl-7.9.1/lib/ftp.c	Sun Nov  4 20:24:56 2001
@@ -200,15 +200,8 @@
   if (ftpcode)
     *ftpcode = 0; /* 0 for errors */
 
-  if(data->set.timeout) {
-    /* if timeout is requested, find out how much remaining time we have */
-    timeout = data->set.timeout - /* timeout time */
-      Curl_tvdiff(Curl_tvnow(), conn->now)/1000; /* spent time */
-    if(timeout <=0 ) {
-      failf(data, "Transfer aborted due to timeout");
-      return -SELECT_TIMEOUT; /* already too little time */
-    }
-  }
+  if(data->set.timeout)
+    timeout = data->set.timeout;
 
   FD_ZERO (&readfd);		/* clear it */
   FD_SET (sockfd, &readfd);     /* read socket */
diff -Nru curl-7.9.1.org/lib/transfer.c curl-7.9.1/lib/transfer.c
--- curl-7.9.1.org/lib/transfer.c	Sat Nov  3 00:09:39 2001
+++ curl-7.9.1/lib/transfer.c	Sun Nov  4 20:22:11 2001
@@ -284,7 +284,7 @@
       writefd = wkeepfd;
       interval.tv_sec = 1;
       interval.tv_usec = 0;
-
+      start = Curl_tvnow();
       switch (select (maxfd, &readfd, &writefd, NULL, &interval)) {
       case -1:			/* select() error, stop reading */
 #ifdef EINTR
