--- curl-8.11.0/acinclude.m4.orig	2024-11-06 08:09:19.000000000 +0100
+++ curl-8.11.0/acinclude.m4	2024-11-15 10:28:27.723007598 +0100
@@ -565,7 +565,7 @@
       else
         LIBS="$curl_cv_ldap_LIBS $curl_cv_save_LIBS"
       fi
-      LIBCURL_PC_REQUIRES_PRIVATE="ldap $LIBCURL_PC_REQUIRES_PRIVATE"
+      #LIBCURL_PC_REQUIRES_PRIVATE="ldap $LIBCURL_PC_REQUIRES_PRIVATE"
       AC_MSG_RESULT([$curl_cv_ldap_LIBS])
       ;;
   esac
